.github/workflows/lint.yml

```diff
@@ -32,6 +46,13 @@ jobs:
       contents: read
     runs-on: ubuntu-latest
     steps:
+      - uses: step-security/harden-runner@main
+        with:
+          allowed-endpoints:
+            agent.api.stepsecurity.io:443
+            github.com:443
+            nodejs.org:443
+            registry.npmjs.org:443
```

Comment:
is there any way an individual step could be annotated rather than broadly allowing every step to access these endpoints?

Reply:
Hi @ljharb , individual steps are annotated in the [web UI](https://app.stepsecurity.io/github/varunsh-coder/nvm/actions/runs/1490899693)/ API (thanks for that idea BTW) and I can detect and alert that a step has called an endpoint it was not supposed to, but technically I cannot stop it from doing so. Configuring the firewall per step would be hard. Let me know if it would be better to just detect and alert at a step level v.s. restricting at a job level.

Reply:
What if i use the runner multiple times, and each time, I define a brand new set of allowed endpoints, that only apply to further steps?

Reply:
Thats an interesting idea. I can experiment with it. It might lead to lot of steps that call `harden-runner` which may make the workflow file look cluttered. It would be ideal if this could be done without the additional steps. Let me think about it.